### DISCUSSION TITLE
What is a chain-aware provider?

### DISCUSSION THREAD

**Discussion by jmrossy** [wevm/wagmi#755] [Q&A] [answered]
Hi, thanks for the great work on this lib!

The docs mention a 'chain-aware' provider in a few places ([example](https://wagmi.sh/docs/hooks/useProvider#configuration)) but I can't find details on what makes a provider chain-aware.

Is it simply a function that takes a chain and returns an Ethers.js provider like in [this comment here](https://github.com/wagmi-dev/wagmi/issues/71#issuecomment-1013748986)?

**Comment by tmm** [comment, marked as answer]
Yup, exactly! It’s able to update the ethers provider when a connected account switches chains.

**Reply by jmrossy** [reply]
Thanks!
